src/gui/entry/EntryView.cpp

```diff
@@ -474,6 +480,8 @@ void EntryView::resetViewToDefaults()
     header()->hideSection(EntryModel::Attachments);
     header()->hideSection(EntryModel::Size);
     header()->hideSection(EntryModel::PasswordStrength);
+    header()->hideSection(EntryModel::Color);
+    m_model->setColorVisible(true);
```

Comment:
Instead of hard coding the state change in various spots in the code, recommend adding a private function such as "onHeaderChanged" and then do the generic check/update in that function. Call that function from all the places you are setting the color visible. `m_model->setColorVisible(isColumnHidden(EntryModel::Color));`

Reply:
Did you mean it like i applied it? It was just called in two places, but its more reusable now.

Reply:
Yah we might expand on this function in the future. I noticed there is no signal from QHeaderView for when a column is hidden/visible changed.